onnxruntime/core/providers/cpu/tensor/concat.cc

```diff
@@ -35,66 +35,117 @@ Status ConcatBase::PrepareForCompute(OpKernelContext* ctx,
                                      const std::vector<const Tensor*>& input_tensors,
                                      Prepare& p) const {
   int input_count = static_cast<int>(input_tensors.size());
+
   // Must have atleast one input to concat
   ORT_RETURN_IF_NOT(input_count >= 1, "Must have 1 or more inputs");
 
-  const Tensor* tensor_pointer = input_tensors[0];
-  if (tensor_pointer == nullptr)
-    return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+  std::vector<int64_t> reference_dims;
+  size_t reference_rank = 0;
+
+  int reference_tensor_index = 0;
+
+  std::vector<int64_t> input_tensor_sizes;
+  input_tensor_sizes.reserve(input_count);
+
+  bool all_inputs_are_empty = true;
+
+  for (int index = 0; index < input_count; ++index) {
+    const auto* input = input_tensors[index];
+    ORT_ENFORCE(input != nullptr, "input count mismatch");
+
```

Comment:
nit: ORT_ENFORCE is somewhat expensive in terms of binary size and this check seems unnecessary. If we want to protect against input_tensors having a nullptr entry it would be better to convert input_tensors to std::vector<std::reference_wrapper<const Tensor>> so it breaks at the calling site instead of here.

Reply:
Yeah this check seems unnecessary. I think it just propagated from the original implementation. I will remove them as part of another change instead of wasting CI resources over this.